src/query/service/src/interpreters/interpreter_copy.rs

```diff
@@ -254,67 +240,120 @@ impl CopyInterpreter {
         &self,
         plan: &CopyIntoTablePlan,
     ) -> Result<PipelineBuildResult> {
+        let catalog = plan.catalog_name.as_str();
+        let database = plan.database_name.as_str();
+        let table = plan.table_name.as_str();
+
         let ctx = self.ctx.clone();
-        let to_table = ctx
-            .get_table(&plan.catalog_name, &plan.database_name, &plan.table_name)
-            .await?;
+        let to_table = ctx.get_table(catalog, database, table).await?;
 
-        let (mut build_res, source_schema, files) = if let Some(query) = &plan.query {
-            let (build_res, source_schema) = self.build_query(query).await?;
-            (
-                build_res,
-                source_schema,
-                plan.stage_table_info
+        let mut build_res;
+        let source_schema;
+        let files;
+
+        match &plan.query {
+            None => {
+                let table_ctx: Arc<dyn TableContext> = self.ctx.clone();
+
+                files = plan.collect_files(&table_ctx).await?;
+                source_schema = plan.required_source_schema.clone();
+                build_res = PipelineBuildResult::create();
+                if !files.is_empty() {
+                    self.build_read_stage_table_data_pipeline(
+                        &mut build_res.main_pipeline,
+                        plan,
+                        to_table.get_block_thresholds(),
+                        files.clone(),
+                    )
+                    .await?;
+                }
```

Comment:
```suggestion
                } else {
                    return Ok(build_res);
                }
```

Reply:
I have set it in line260: `build_res = PipelineBuildResult::create();`

Reply:
I see, thanks.